network-config-analyzer/ResourcesHandler.py

```diff
@@ -0,0 +1,182 @@
+        """
+        Chooses to parse the namespaces from the ns_list if exists in the input resources.
+        the pods and services: from pod_list if exist in the input resources
+        if the specified lists do not exist, parses them from the resources list
+        """
+        if ns_list:
+            ns_resource = ns_list
+            if resource_list:
+                print('Warning: namespaces will be taken from the provided namespace list key.'
+                      'input resources provided with resource list key will be ignored when finding namespaces')
+        else:
+            ns_resource = resource_list
+
+        if pod_list:
+            pod_resource = pod_list
+            if resource_list:
+                print('Warning: pods and services will be taken from the provided pod list key. '
+                      'input resources provided with resource list key will be ignored when finding pods and services')
+        else:
+            pod_resource = resource_list
+
+        if pod_resource is None or ns_resource is None:
+            return False  # can not build peer container
+
+        if pod_resource == resource_list and pod_resource == ns_resource:  # both in resourceList
+            print(f'Finding topology objects in {resource_list}')
+            self._parse_resources_path(resource_list, np_flag=False, ns_flag=True, pod_flag=True)
+        else:  # we always want to parse for ns first
+            self._parse_resources_path(ns_resource, ns_flag=True)
+            self._parse_resources_path(pod_resource, pod_flag=True)
+
+        if len(self.pods_finder.peer_set) == 0 or len(self.ns_finder.namespaces) == 0:
+            return False  # can not build peer container
+
+        print(f'{config_name}: cluster has {len(self.pods_finder.peer_set)} unique endpoints, '
+              f'{len(self.ns_finder.namespaces)} namespaces')
+        return True
+
+    def parse_lists_for_policies(self, np_list, resource_list, peer_container):
+        """
+        parses policies from np_list resource if exists, otherwise parses the resource_list for policies
+        :param np_list: list of entries of networkPolicies
+        :param resource_list: list of entries
+        :param peer_container: the existing PeerContainer
+        """
+        self.policies_finder.set_peer_container(peer_container)
+        if np_list and np_list != ['']:
+            self._parse_resources_path(np_list, np_flag=True)
+            if resource_list:
+                print('Warning: policies will be taken from the provided networkPolicy list key. '
+                      'input resources provided with resource list key will be ignored when finding network policies')
+        elif resource_list:
+            print(f'Finding policies in {resource_list}')
+            self._parse_resources_path(resource_list, np_flag=True)
+
+    def _parse_resources_path(self, resource_list, np_flag=False, ns_flag=False, pod_flag=False):
```

Comment:
Instead of using multiple input flags, can define an enumerated type of resource type to parse, and send a single value of this type.